launcher/internal/signaturediscovery/client_test.go

```diff
@@ -48,17 +48,46 @@ func TestFormatSigTag(t *testing.T) {
 
 func TestFetchSignedImageManifestDockerPublic(t *testing.T) {
 	ctx := namespaces.WithNamespace(context.Background(), "test")
-	containerdClient, err := containerd.New(defaults.DefaultAddress)
-	if err != nil {
-		t.Skipf("test needs containerd daemon: %v", err)
-	}
-	defer containerdClient.Close()
 
 	targetRepository := "gcr.io/distroless/static"
 	originalImageDesc := v1.Descriptor{Digest: "sha256:9ecc53c269509f63c69a266168e4a687c7eb8c0cfd753bd8bfcaa4f58a90876f"}
-	client := New(containerdClient, originalImageDesc)
+	client := createTestClient(t, originalImageDesc)
 	// testing image manifest fetching using a public docker repo url
 	if _, err := client.FetchSignedImageManifest(ctx, targetRepository); err != nil {
 		t.Errorf("failed to fetch signed image manifest from targetRepository [%s]: %v", targetRepository, err)
 	}
 }
+
+func TestFetchImageSignaturesDockerPublic(t *testing.T) {
+	ctx := namespaces.WithNamespace(context.Background(), "test")
+	originalImageDesc := v1.Descriptor{Digest: "sha256:9ecc53c269509f63c69a266168e4a687c7eb8c0cfd753bd8bfcaa4f58a90876f"}
+	targetRepository := "gcr.io/distroless/static"
+
+	client := createTestClient(t, originalImageDesc)
+	signatures, err := client.FetchImageSignatures(ctx, targetRepository)
+	if err != nil {
+		t.Errorf("failed to fetch image signatures from targetRepository [%s]: %v", targetRepository, err)
+	}
+	if len(signatures) == 0 {
+		t.Errorf("no image signatures found for the original image %v", originalImageDesc)
+	}
+	for _, sig := range signatures {
+		if _, err := sig.Payload(); err != nil {
+			t.Errorf("Payload() failed: %v", err)
+		}
+		if _, err := sig.Base64Encoded(); err != nil {
+			t.Errorf("Base64Encoded() failed: %v", err)
+		}
+	}
+}
+
+func createTestClient(t *testing.T, originalImageDesc v1.Descriptor) *Client {
+	t.Helper()
+
+	containerdClient, err := containerd.New(defaults.DefaultAddress)
+	if err != nil {
+		t.Skipf("test needs containerd daemon: %v", err)
```

Comment:
(this can be a followup) Can we test this in CI https://github.com/google/go-tpm-tools/blob/master/.github/workflows/ci.yml

Reply:
For this specific test, it has to be run under root permission. Will add a new entry to CI to run this test under root only in my followup PR.